akka-stream-typed/src/main/scala/akka/stream/typed/scaladsl/ActorFlow.scala

```diff
@@ -128,4 +128,48 @@ object ActorFlow {
     askFlow
   }
 
+  /**
+   * Use for messages whose response is known to be a [[akka.pattern.StatusReply]]. When a [[akka.pattern.StatusReply#success]] response
+   * arrives the future is completed with the wrapped value, if a [[akka.pattern.StatusReply#error]] arrives the future is instead
+   * failed.
+   */
+  def askWithStatus[I, Q, A](ref: ActorRef[Q])(makeMessage: (I, ActorRef[A]) => Q)(
+      implicit timeout: Timeout): Flow[I, A, NotUsed] =
+    askWithStatus(2)(ref)(makeMessage)
+
+  /**
+   * Use for messages whose response is known to be a [[akka.pattern.StatusReply]]. When a [[akka.pattern.StatusReply#success]] response
+   * arrives the future is completed with the wrapped value, if a [[akka.pattern.StatusReply#error]] arrives the future is instead
+   * failed.
+   */
+  def askWithStatus[I, Q, A](parallelism: Int)(ref: ActorRef[Q])(makeMessage: (I, ActorRef[A]) => Q)(
```

Comment:
Couldn't it be composed with the existing ActorFlow + map instead of another implementation? Something like:


```
ActorFlow.ask[I, Q, A](parallelism: Int)(ref: ActorRef[Q])(makeMessage: (I, ActorRef[StatusReply[A]]) => Q) = 
  actorFlow.ask(int)(ref)(makeMessage).map {
  case StatusReply.Success(a) => a
  case StatusReply.Error(err) => throw err
}
```

Reply:
Indeed. Done